src/screens/cart-screen.tsx

```diff
@@ -1,10 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Linking } from 'react-native';
 import { Box, Button, ScrollView, Flex, Text } from 'native-base';
 import { CartItem } from 'src/cards/cart-item';
 import { useAppSelector } from 'src/hooks/useful-ducks';
+import { db } from 'src/firebase/firebase-config';
+import { loadStripe } from '@stripe/stripe-js';
+import { collection, addDoc, onSnapshot, DocumentData, DocumentSnapshot } from 'firebase/firestore';
+import { string } from 'yup';
+
+interface Data {
+    error: any;
+    url: string;
+}
 
 export const CartScreen = () => {
+    const [loading, setLoading] = useState(false);
+
     const cart = useAppSelector((state) => state.cart);
+    const user = useAppSelector((state) => state.user);
+
+    const checkOut = async () => {
+        // Linking.openURL('https://developer.mozilla.org/en-US/');
+
+        console.log(user.uid);
+        setLoading(true);
+
+        const userCollection = collection(db, 'users', user.uid, 'checkout_sessions');
+
+        const docRef = await addDoc(userCollection, {
```

Comment:
Remove the firebase calls in the screens - need to make some sort of middleware query function instead using rtk-query